File: speech-synthesis/amazon-polly/AmazonyPollyTextToSpeech.java
// "ssml" is the input provided in the Speech Synthesis Markup Language.
// Alternatively, you can provide raw text and set the text type to "TextType.Text".

AmazonPolly client = AmazonPollyClientBuilder
    .standard()
    .withCredentials(credentialsProvider)
    .withRegion(Regions.US_EAST_1)
    .build();

SynthesizeSpeechRequest request = new SynthesizeSpeechRequest()
    .withText(ssml)
    .withTextType(TextType.Ssml)
    .withVoiceId(VoiceId.Matthew)
    .withOutputFormat(OutputFormat.Mp3);

SynthesizeSpeechResult result = client.synthesizeSpeech(request);

InputStream audioStream = result.getAudioStream();

FileOutputStream fileOutputStream = new FileOutputStream(outputFile);

IOUtils.copy(audioStream, fileOutputStream);

audioStream.close();
fileOutputStream.close();

File: speech-synthesis/amazon-polly/AmazonyPollyTextToSpeech.kt
// "ssml" is the input provided in the Speech Synthesis Markup Language.
// Alternatively, you can provide raw text and set the text type to "TextType.Text".

val client = AmazonPollyClientBuilder
    .standard()
    .withCredentials(credentialsProvider)
    .withRegion(Regions.US_EAST_1)
    .build()

val request = SynthesizeSpeechRequest()
    .withText(ssml)
    .withTextType(TextType.Ssml)
    .withVoiceId(VoiceId.Matthew)
    .withOutputFormat(OutputFormat.Mp3)

val result = client.synthesizeSpeech(request)
val audioStream = result.audioStream

val bytes = audioContent.toByteArray()
FileOutputStream(outputFile).use { stream -> stream.write(bytes) }
